build/rmkdepend/include.c

```diff
@@ -299,11 +299,11 @@ struct inclist *inc_path(char *file, char *include, boolean dot) {
     */
    if (!found)
       for (pp = includedirs; *pp; pp++) {
-         if (strlen(*pp) + strlen(include) + 2 > sizeof(path)) {
+         int retlen = snprintf(path, BUFSIZ, "%s/%s", *pp, include);
+         if (retlen <= 0 || BUFSIZ <= retlen) {
```

Comment:
```suggestion
         if (retlen <= 0 || BUFSIZ < retlen) {
```
The return value does not count/include [the terminating null character](https://cplusplus.com/reference/cstdio/snprintf/) .  Also a negative return value indicates an encoding error rather than a length problem (so it probably does not matter and it should never happen here)